src/util.ts

```diff
@@ -0,0 +1,104 @@
+import { Keyring, WsProvider } from "@polkadot/api";
+import { cryptoWaitReady } from "@polkadot/util-crypto";
+import assert from "assert";
+
+import { OPENGOV_LARGE_TIP_VALUE, OPENGOV_MEDIUM_TIP_VALUE, OPENGOV_SMALL_TIP_VALUE } from "./constants";
+import { ContributorAccount, OpenGovTrack, State, TipMetadata, TipNetwork, TipRequest, TipSize } from "./types";
+
+const validTipSizes: { [key: string]: TipSize } = { small: "small", medium: "medium", large: "large" } as const;
+
+export function getTipSize(tipSizeInput: string | undefined): TipSize {
+  if (!tipSizeInput || !(tipSizeInput in validTipSizes)) {
+    throw new Error(`Invalid tip size. Please specify one of ${Object.keys(validTipSizes).join(", ")}.`);
+  }
+
+  return validTipSizes[tipSizeInput];
+}
+
+export function tipSizeToOpenGovTrack(tipSize: TipSize): { track: OpenGovTrack; value: number } {
+  if (tipSize === "small") return { track: "SmallTipper", value: OPENGOV_SMALL_TIP_VALUE };
+  if (tipSize === "medium") return { track: "BigTipper", value: OPENGOV_MEDIUM_TIP_VALUE };
+  if (tipSize === "large") return { track: "BigTipper", value: OPENGOV_LARGE_TIP_VALUE };
+
+  throw new Error(`Invalid tip size. Please specify one of ${Object.keys(validTipSizes).join(", ")}.`);
+}
+
+export function parseContributorAccount(pullRequestBody: string | null): ContributorAccount {
+  const matches =
+    typeof pullRequestBody === "string" &&
+    pullRequestBody.match(
+      // match "polkadot address: <ADDRESS>"
+      /(\S+)\s*address:\s*([a-z0-9]+)/i,
```

Comment:
Nice

Reply:
This is already-existing code, moved around.